gnomad_qc/v3/variant_qc/evaluation.py

```diff
@@ -0,0 +1,328 @@
+        ht = get_filters(model_id, split=True).ht()
+
+        ht = ht.filter(
+            ~info_ht[ht.key].AS_lowqual
+        )
+        ht = ht.annotate(
+            **rf_ht[ht.key],
+            info=info_ht[ht.key].info,
+            score=ht.info.AS_VQSLOD,
+            positive_train_site=ht.info.POSITIVE_TRAIN_SITE,
+            negative_train_site=ht.info.NEGATIVE_TRAIN_SITE,
+            AS_culprit=ht.info.AS_culprit,
+        )
+
+    else:
+        ht = get_rf_result(model_id=model_id).ht()
+        ht = ht.annotate(
+            info=info_ht[ht.key].info,
+            positive_train_site=ht.tp,
+            negative_train_site=ht.fp,
+            score=ht.rf_probability["TP"],
+        )
+
+    ht = ht.filter(ht.ac_raw > 0)
+    ht = ht.filter(
+        ~info_ht[ht.key].AS_lowqual & ~hl.is_defined(telomeres_and_centromeres.ht()[ht.locus])
+    )
+    ht_lcr = filter_low_conf_regions(
+        ht,
+        filter_lcr=True,
+        # TODO: Uncomment when we have decoy path
+        filter_decoy=False,  # True,
+        filter_segdup=True,
+    )
+    ht = ht.annotate(non_lcr=hl.is_defined(ht_lcr[ht.key]))
+    bin_ht = create_binned_ht(ht, n_bins, add_substrat={"non_lcr": ht.non_lcr})
+    bin_ht.write(
+        get_score_quantile_bins(model_id, aggregated=False).path, overwrite=overwrite
+    )
+
+
+def create_grouped_bin_ht(model_id: str, overwrite: bool = False) -> None:
+    """
+    Creates binned data from a quantile bin annotated Table grouped by bin_id (rank, bi-allelic, etc.), contig, snv,
+    bi_allelic and singleton containing the information needed for evaluation plots.
+
+    :param str model_id: Which variant QC model (rf or vqsr model ID) to group
+    :param bool overwrite: Should output files be overwritten if present
+    :return: None
+    :rtype: None
+    """
+
+    ht = get_score_quantile_bins(model_id, aggregated=False).ht()
+
+    # Count variants for ranking
+    count_expr = {
+        x: hl.agg.filter(
+            hl.is_defined(ht[x]),
+            hl.agg.counter(
+                hl.cond(hl.is_snp(ht.alleles[0], ht.alleles[1]), "snv", "indel")
+            ),
+        )
+        for x in ht.row
+        if x.endswith("bin")
+    }
+    bin_variant_counts = ht.aggregate(hl.struct(**count_expr))
+    logger.info(f"Found the following variant counts:\n {pformat(bin_variant_counts)}")
+    ht = ht.annotate_globals(bin_variant_counts=bin_variant_counts)
+
+    # Load ClinVar pathogenic data
+    clinvar_pathogenic_ht = filter_to_clinvar_pathogenic(clinvar.ht())
+    ht = ht.annotate(clinvar_path=hl.is_defined(clinvar_pathogenic_ht[ht.key]))
+    trio_stats_ht = fam_stats.ht()
+
+    logger.info(f"Creating grouped bin table...")
+    grouped_binned_ht = compute_grouped_binned_ht(
+        ht, checkpoint_path=get_checkpoint_path(f"grouped_bin_{model_id}"),
+    )
+
+    logger.info(f"Aggregating grouped bin table...")
+    parent_ht = grouped_binned_ht._parent
+    agg_ht = grouped_binned_ht.aggregate(
```

Comment:
If you try to use the grouped HT not the parent HT you get the following error ```TypeError: 'GroupedTable' object is not subscriptable at this line indel_length=hl.abs(ht.alleles[0].length() - ht.alleles[1].length())``` because the GroupedTable gets aggregated with expressions from the Table it comes from so you need to get the parent table